sdk/attestation/azure-security-attestation/src/private/attestation_deserializers_private.cpp

```diff
@@ -0,0 +1,297 @@
+    return result;
+  }
+
+  AttestationTokenHeader AttestationTokenHeaderSerializer::Deserialize(
+      Azure::Core::Json::_internal::json const& jsonHeader)
+  {
+    AttestationTokenHeader returnValue;
+
+    JsonOptional::SetIfExists(returnValue.Algorithm, jsonHeader, "alg");
+    JsonOptional::SetIfExists(returnValue.KeyURL, jsonHeader, "jku");
+
+    if (jsonHeader.contains("jwk"))
+    {
+      auto jwk = JsonWebKeySerializer::Deserialize(jsonHeader["jwk"]);
+      returnValue.Key = AttestationSignerInternal(jwk);
+    }
+    JsonOptional::SetIfExists(returnValue.KeyId, jsonHeader, "kid");
+    JsonOptional::SetIfExists(returnValue.X509Url, jsonHeader, "x5u");
+    JsonOptional::SetIfExists(returnValue.X509CertificateChain, jsonHeader, "x5c");
+    JsonOptional::SetIfExists(returnValue.CertificateSha256Thumbprint, jsonHeader, "x5t#S256");
+    JsonOptional::SetIfExists(returnValue.CertificateThumbprint, jsonHeader, "x5t");
+    JsonOptional::SetIfExists(returnValue.Type, jsonHeader, "typ");
+    JsonOptional::SetIfExists(returnValue.ContentType, jsonHeader, "cty");
+    JsonOptional::SetIfExists(returnValue.Critical, jsonHeader, "crit");
+    return returnValue;
+  }
+  std::string AttestationTokenHeaderSerializer::Serialize(AttestationTokenHeader const& tokenHeader)
+  {
+    Azure::Core::Json::_internal::json serializedHeader;
+    JsonOptional::SetFromNullable(tokenHeader.Algorithm, serializedHeader, "alg");
+    JsonOptional::SetFromNullable(tokenHeader.KeyURL, serializedHeader, "jku");
+    JsonOptional::SetFromNullable<AttestationSigner, std::string>(
+        tokenHeader.Key, serializedHeader, "jwk", [](AttestationSigner const& signer) {
+          return AttestationSignerInternal::SerializeToJson(signer);
+        });
+    JsonOptional::SetFromNullable(tokenHeader.ContentType, serializedHeader, "cty");
+    JsonOptional::SetFromNullable(tokenHeader.Critical, serializedHeader, "crit");
+    JsonOptional::SetFromNullable(tokenHeader.KeyId, serializedHeader, "kid");
+    JsonOptional::SetFromNullable(tokenHeader.Type, serializedHeader, "typ");
+
+    JsonOptional::SetFromNullable(tokenHeader.X509CertificateChain, serializedHeader, "x5c");
+    JsonOptional::SetFromNullable(tokenHeader.X509Url, serializedHeader, "x5u");
+    JsonOptional::SetFromNullable(
+        tokenHeader.CertificateSha256Thumbprint, serializedHeader, "x5t#S256");
+    JsonOptional::SetFromNullable(tokenHeader.CertificateThumbprint, serializedHeader, "x5t");
+
+    return serializedHeader.dump();
+  }
+
+  JsonWebKey JsonWebKeySerializer::Deserialize(const Azure::Core::Json::_internal::json& jwk)
+  {
+    JsonWebKey returnValue;
+    Azure::Nullable<std::string> kty;
+    JsonOptional::SetIfExists(kty, jwk, "kty");
+    if (kty.HasValue())
+    {
+      returnValue.kty = kty.Value();
+    }
+    else
+    {
+      throw std::runtime_error("Could not find required field 'kty' in JSON Web Key");
+    }
```

Comment:
We can avoid creating the temporal value here:

```suggestion

    JsonOptional::SetIfExists(returnValue.kty, jwk, "kty");
    
    // returnValue.kty gets initialized to default (no value) and won't be changed by SetIfExists if kty is not found
    if (!returnValue.kty)
    {
      throw std::runtime_error("Could not find required field 'kty' in JSON Web Key");
    }
```